cmd/forwarder/fixer.go

```diff
@@ -65,7 +65,8 @@ func getMetadata(req *http.Request, cred *credential, metadataId string) ([]byte
 // Write a header field into the messageWriter buffer. Truncates to maxLength
 // Returns true if the input string was truncated, and false otherwise.
 func writeField(messageWriter *bytes.Buffer, str []byte, maxLength int) bool {
-	if maxLength > 48 {
+	length := len(string(str))
+	if length > maxLength {
 		messageWriter.Write(str[0:maxLength])
```

Comment:
Be careful with this: https://play.golang.org/p/CM54_Nfv3Fj

Reply:
PS: This may not be an issue if you know there is zero chance of unicode characters.

Reply:
@freeformz Hi Ed! :D 

All of the fields we're truncating here are from the `HEADER`. They're all defined as being `PRINTUSASCII` characters, see section 6.1 here https://tools.ietf.org/html/rfc5424 (I'm sure you're familiar with this document, given your history with Logplex). So I think we're safe in not worrying about unicode here.